pkg/plugins/generic/generic_plugin.go

```diff
@@ -217,35 +262,46 @@ func needDrainNode(desired sriovnetworkv1.Interfaces, current sriovnetworkv1.Int
 	return
 }
 
-func needRebootNode(state *sriovnetworkv1.SriovNetworkNodeState, loadVfioDriver *uint, loadVirtioVdpaDriver *uint) (needReboot bool) {
-	needReboot = false
-	if *loadVfioDriver != loaded {
-		if needVfioDriver(state) {
-			*loadVfioDriver = loading
-			update, err := tryEnableIommuInKernelArgs()
-			if err != nil {
-				glog.Errorf("generic-plugin needRebootNode():fail to enable iommu in kernel args: %v", err)
-			}
-			if update {
-				glog.V(2).Infof("generic-plugin needRebootNode(): need reboot for enabling iommu kernel args")
-			}
-			needReboot = needReboot || update
+func needRebootIfVfio(state *sriovnetworkv1.SriovNetworkNodeState, driverMap DriverStateMapType) (needReboot bool) {
+	driverState := driverMap[Vfio]
+	if !driverState.DriverLoaded && driverState.NeedDriverFunc(state, driverState) {
+		var err error
+		needReboot, err = tryEnableIommuInKernelArgs()
+		if err != nil {
+			glog.Errorf("generic-plugin needRebootNode():fail to enable iommu in kernel args: %v", err)
 		}
-	}
-
-	if *loadVirtioVdpaDriver != loaded {
-		if needVirtioVdpaDriver(state) {
-			*loadVirtioVdpaDriver = loading
+		if needReboot {
+			glog.V(2).Infof("generic-plugin needRebootNode(): need reboot for enabling iommu kernel args")
 		}
 	}
+	return needReboot
+}
+
+func needRebootNode(state *sriovnetworkv1.SriovNetworkNodeState, driverMap DriverStateMapType) (needReboot bool) {
+	needReboot = needRebootIfVfio(state, driverMap)
+	updateNode, err := utils.WriteSwitchdevConfFile(state)
 
-	update, err := utils.WriteSwitchdevConfFile(state)
 	if err != nil {
 		glog.Errorf("generic-plugin needRebootNode(): fail to write switchdev device config file")
 	}
-	if update {
+	if updateNode {
 		glog.V(2).Infof("generic-plugin needRebootNode(): need reboot for updating switchdev device configuration")
 	}
-	needReboot = needReboot || update
+	needReboot = needReboot || updateNode
 	return
 }
+
+// ////////////// for testing purposes only ///////////////////////
+func (p *GenericPlugin) GetDriverStateMap() DriverStateMapType {
```

Comment:
nit: Can you make these two functions package-private?

```
func (p *GenericPlugin) getDriverStateMap() DriverStateMapType {

func (p *GenericPlugin) loadDriverForTests(state *sriovnetworkv1.SriovNetworkNodeState) {
```

Reply:
Had to change the test package as well. Thanks!